test/v2/controllers/CreateUkPropertyPeriodSummaryControllerSpec.scala

```diff
@@ -0,0 +1,455 @@
+        status(result) shouldBe CREATED
+        contentAsJson(result) shouldBe responseBody
+        header("X-CorrelationId", result) shouldBe Some(correlationId)
+
+
+        val auditResponse: AuditResponse = AuditResponse(OK, None, Some(hateoasResponse))
+        MockedAuditService.verifyAuditEvent(consolidatedEvent(auditResponse)).once
+      }
+    }
+
+    "return a successful response from an unconsolidated request" when {
+      "the request received is valid" in new Test {
+
+        MockCreateUkPropertyRequestParser
+          .requestFor(CreateUkPropertyPeriodSummaryRawData(nino, taxYear, businessId, requestBodyJson))
+          .returns(Right(requestData))
+
+        MockCreateUkPropertyService
+          .createUkProperty(requestData)
+          .returns(Future.successful(Right(ResponseWrapper(correlationId, response))))
+
+        MockHateoasFactory
+          .wrap(response, CreateUkPropertyPeriodSummaryHateoasData(nino, businessId, submissionId))
+          .returns(HateoasWrapper(response, Seq.empty))
+
+        val result: Future[Result] = controller.handleRequest(nino, businessId, taxYear)(fakePostRequest(requestBodyJson))
+        status(result) shouldBe CREATED
+        contentAsJson(result) shouldBe responseBody
+        header("X-CorrelationId", result) shouldBe Some(correlationId)
+
+        val auditResponse: AuditResponse = AuditResponse(OK, None, Some(hateoasResponse))
+        MockedAuditService.verifyAuditEvent(unconsolidatedEvent(auditResponse)).once
+      }
+    }
+
+    "return the error as per spec" when {
+      "parser errors occur" must {
+        def errorsFromParserTester(error: MtdError, expectedStatus: Int): Unit = {
+          s"a ${error.code} error is returned from the parser" in new Test {
+
+            MockCreateUkPropertyRequestParser
+              .requestFor(rawData)
+              .returns(Left(ErrorWrapper(correlationId, error, None)))
+
+            val result: Future[Result] = controller.handleRequest(nino, businessId, taxYear)(fakePostRequest(requestBodyJson))
+
+            status(result) shouldBe expectedStatus
+            contentAsJson(result) shouldBe Json.toJson(error)
+            header("X-CorrelationId", result) shouldBe Some(correlationId)
+
+            val auditResponse: AuditResponse = AuditResponse(expectedStatus, Some(Seq(AuditError(error.code))), None)
+            MockedAuditService.verifyAuditEvent(unconsolidatedEvent(auditResponse)).once
+          }
+        }
+
+        val input = Seq(
+          (BadRequestError, BAD_REQUEST),
+          (TaxYearFormatError, BAD_REQUEST),
+          (NinoFormatError, BAD_REQUEST),
+          (BusinessIdFormatError, BAD_REQUEST),
+          (RuleTaxYearRangeInvalidError, BAD_REQUEST),
+          (RuleTaxYearNotSupportedError, BAD_REQUEST),
+          (ToDateFormatError, BAD_REQUEST),
+          (FromDateFormatError, BAD_REQUEST),
+          (RuleBothExpensesSuppliedError, BAD_REQUEST),
```

Comment:
Add a `paths` here (as well as in `RuleIncorrectOrEmptyBodyError` case) so that we fail if the controller matches on the full error object.

Reply:
I just followed foreign test scenarios

Reply:
it is covered in ITs

Reply:
Ok